src/goto-symex/goto_trace.cpp

```diff
@@ -360,13 +385,44 @@ void correctness_graphml_goto_trace(
   graph.generate_graphml(options);
 }
 
+void appendInfo(
+  std::string &dest,
+  const std::string &label,
+  const std::string &value)
+{
+  if (!value.empty())
+  {
+    if (!dest.empty())
+      dest += " ";
+    dest += label + " " + id2string(value);
+  }
+}
+
+void show_simplified_location(std::ostream &out, const locationt &location)
+{
+  std::string dest;
+  const irep_idt &file = location.get_file();
+  const irep_idt &line = location.get_line();
+  const irep_idt &function = location.get_function();
+
+  if (file != "")
+    appendInfo(dest, "file", id2string(file));
+  if (line != "")
+    appendInfo(dest, "line", id2string(line));
+  if (function != "")
+    appendInfo(dest, "function", id2string(function));
+  out << dest << "\n";
+}
+
 void show_goto_trace(
   std::ostream &out,
   const namespacet &ns,
   const goto_tracet &goto_trace)
 {
   unsigned prev_step_nr = 0;
   bool first_step = true;
+  bool cex_only = config.options.get_bool_option("cex-only");
+  bool simplify_trace = config.options.get_bool_option("simplify-trace");
 
   for (const auto &step : goto_trace.steps)
   {
```

Comment:
Can't we simplify the code to this?
This should allow us to remove all those `if(!cex_only)` statements.
```suggestion
  for (const auto &step : goto_trace.steps)
  {
    // we only care about the counter example, which is only triggered by assert steps. Ignore all other steps.
    if (cex_only && step.type != goto_trace_stept::ASSERT)
      continue;
```

Reply:
Changed as suggested